### DISCUSSION TITLE
Opening a packaged html file on Android

### DISCUSSION THREAD

**Discussion by XiaoGen-Taiwan** [airsdk/Adobe-Runtime-Support#2737] [Q&A]
Hi,

I'm trying to deliver help files (as html) in my bundled application. 
I include a 'help' folder in the package. 
I'm calling it as following:
```
private function onOpenHelp():void {
   try {
	var help_file:String = "help" + File.separator + "locales" + File.separator + resourceManager.localeChain[0] + File.separator + "help.html";
	var file:File = File.applicationDirectory.resolvePath(help_file);
		
	if (t_file && t_file.exists) {
		var t_urlRequest:URLRequest = new URLRequest(t_file.url);
		navigateToURL(t_urlRequest, "_self");
	}
		
   } catch (e:Error) {
	trace("Help not available");
   }
}
```
On win11 it works but on Android it says 'Requested URL is not available'. 
Anyone an idea?

Additionally: 
1) Is there a way to jump to a specific anchor (e.g. .../help/help.html#anchor)? 
2) Is there a way around that damn CORS shit (sorry but that makes me really pissed)
... I can not even call another packaged html page.

Any help would be much appreciated.

**Comment by ajwfrost** [comment]
Might be better to load the URL into a StageWebView instance and keep it within AIR? I think you're hitting the challenge with scoped storage: the browser wouldn't be able to access a file that's part of your application package. You could try perhaps with the `File.openWithDefaultApplication` but that sometimes has issues with these sorts of permissions.

thanks

**Reply by XiaoGen-Taiwan** [reply]
Hi @ajwfrost,
thank you very much again for your quick replys ... you are just amazing.

I tried ```File.openWithDefaultApplication``` but didn't work.

I will try to copy the whole 'help' folder to ```File.applicationStorageDirectory``` at application start
and check if that works. 

If not, I will try the StageWebView approach ... but that might be another challenge.

**Reply by ajwfrost** [reply]
I think `applicationStorageDirectory` would have the same problem. You need a way to give the browser the permissions for opening files that are part of your own application.. short of setting up a local webserver as part of your app, this is going to be a challenge - because thinking about it, even if you used `openWithDefaultApplication`, this would only grant the permission for the first file, and being HTML I assume there are multiple files/images/etc involved in displaying the help.

It may be possible to copy your help files from the application to the documents location using a content resolver, e.g. the first answer under https://stackoverflow.com/questions/67759437/android-11-save-a-file-that-can-other-apps-can-access
But of course, this is also Android version specific.. so there may need to be different handling for the different version.

Which may mean the simplest option is the webview after all......

**Reply by XiaoGen-Taiwan** [reply]
Naaaaa, these Android clowns went faaaaar to faaaaar with their security safety whatever paranoia (mho) ... I can not copy the help folder to anywhere, can not open a html file with webView.loadURL and reading the html file and assign it to webView.loadString will not take css, images, .... into account. Where is this world going to?
